clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -2711,6 +2711,8 @@ static void CollectArgsForIntegratedAssembler(Compilation &C,
   }
   if (!UseRelaxRelocations)
     CmdArgs.push_back("-mrelax-relocations=no");
+  if (Args.hasArg(options::OPT_msse2avx))
+    CmdArgs.push_back("-msse2avx");
```

Comment:
To address @MaskRay target specific comment. Probably you need to put it under sth like `case llvm::Triple::x86 : case llvm::Triple::x86_64:`

Reply:
Done